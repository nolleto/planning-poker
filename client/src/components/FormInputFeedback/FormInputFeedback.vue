<template>
  <div
    :class="{
      [bootstrap.validFeedback]: state === 'valid',
      [bootstrap.invalidFeedback]: state === 'invalid'
    }"
  >
    <slot></slot>
  </div>
</template>

<script>
import styles from '@/styles'

export default {
  name: 'FormInputFeedback',

  props: {
    state: {
      type: String
    }
  },

  computed: {
    bootstrap: () => styles.bootstrap
  }
}
</script>
