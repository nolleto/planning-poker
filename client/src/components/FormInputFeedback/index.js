export { default } from './FormInputFeedback.vue'
