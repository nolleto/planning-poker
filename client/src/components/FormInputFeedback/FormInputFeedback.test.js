import FormInputFeedback from './index'
import { shallowMount } from '@vue/test-utils'

const shallowComponent = () => {
  return shallowMount(FormInputFeedback)
}

describe('FormInputFeedback component', () => {
  let wrapper

  describe('Snapshots', () => {
    beforeAll(() => {
      wrapper = shallowComponent()
    })

    it('matches', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    describe('When has "invalid" state', () => {
      beforeAll(() => {
        wrapper.setProps({ state: 'invalid' })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('When has "valid" state', () => {
      beforeAll(() => {
        wrapper.setProps({ state: 'valid' })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
