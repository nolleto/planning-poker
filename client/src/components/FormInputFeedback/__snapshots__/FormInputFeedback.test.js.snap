// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputFeedback component Snapshots When has "invalid" state matches 1`] = `
<div
  class="invalidFeedback"
/>
`;

exports[`FormInputFeedback component Snapshots When has "valid" state matches 1`] = `
<div
  class="validFeedback"
/>
`;

exports[`FormInputFeedback component Snapshots matches 1`] = `
<div
  class=""
/>
`;
